packages/flutter/lib/src/services/asset_manifest.dart

```diff
@@ -21,6 +23,14 @@ abstract class AssetManifest {
   /// Loads asset manifest data from an [AssetBundle] object and creates an
   /// [AssetManifest] object from that data.
   static Future<AssetManifest> loadFromAssetBundle(AssetBundle bundle) {
+    if (kIsWeb) {
+      return bundle.loadStructuredData(_kAssetManifestWebFilename, (String jsonData) async {
+        // Decode the manifest JSON file to the underlying BIN, and convert to ByteData.
+        final ByteData message = ByteData.sublistView(base64.decode(json.decode(jsonData) as String));
+        // Now we can keep operating as usual.
+        return _AssetManifestBin.fromStandardMessageCodecMessage(message);
```

Comment:
As a quick sanity check, can you use `FlutterTimeline` or something to time how long it takes to decode `jsonData` into an instance of `_AssetManifestBin`?

Reply:
Definitely. I've also redeployed the app in the description in `--profile` mode so we can look at the flame chart from chrome.